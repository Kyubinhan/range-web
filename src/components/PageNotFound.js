import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { RANGE_USE_PLANS } from '../constants/routes';
import { COW_PIC_SRC } from '../constants/variables';

const propTypes = {
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
};

class PageNotFound extends Component {
  componentDidMount() {
    setTimeout(() => {
      this.props.history.push(RANGE_USE_PLANS);
    }, 10000);
  }

  render() {
    return (
      <div className="page-not-found">
        <div className="page-not-found__container">
          <img
            className="page-not-found__image"
            src={COW_PIC_SRC}
            alt="cow-img"
          />
          <div className="page-not-found__title">Page Not Found</div>
          <div className="page-not-found__content">
            <p>This is not the web page you are looking for.</p>
            <p>You will be redirected to the My Range Application home page within 10 seconds.</p>
          </div>
          <div className="page-not-found__link">
            <Link to={RANGE_USE_PLANS}>
              Go to home
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

PageNotFound.propTypes = propTypes;
export default PageNotFound;
