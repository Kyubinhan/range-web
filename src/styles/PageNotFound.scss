// block
.page-not-found {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;

  &__container {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  &__image {
    margin-left: 30px;
    width: 150px;
  }

  &__title {
    text-align: center;
    font-size: 3rem;
    margin: 10px 0 30px 0;
  }

  &__content {
    font-size: 1.3rem;
    text-align: center;
    line-height: 140%;
    p {
      margin: 0;
    }
  }

  &__link {
    margin: 10px auto;
    width: fit-content;
    font-size: 1.3rem;

    &--boldered {
      border: 1px solid  
    }
  }
}