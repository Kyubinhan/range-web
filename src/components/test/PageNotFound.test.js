import React from 'react';
import { shallow } from 'enzyme';
import PageNotFound from '../PageNotFound';

const props = {};
const setupProps = () => {
  props.history = {
    push: jest.fn()
  };
};

beforeEach(() => {
  setupProps();
});


describe('LandingPage', () => {
  xit('renders correctly', () => {
    const wrapper = shallow(<PageNotFound {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Life cycles', () => {
    it('componentDidMount calls the right functions', () => {
      jest.useFakeTimers()
      const wrapper = shallow(<PageNotFound {...props} />);
      expect(props.history.push).not.toBeCalled();

      jest.runAllTimers();
      expect(props.history.push).toHaveBeenCalledTimes(1)
    });
  });
});
