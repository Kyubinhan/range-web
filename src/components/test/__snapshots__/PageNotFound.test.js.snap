// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFound
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="page-not-found__title"
        >
                Error 404
        </div>
        <div
                className="page-not-found__content"
        >
                Woops. Looks like this page doesn't exist 
        </div>
        <div
                className="page-not-found__link"
        >
                <Link
                        replace={false}
                        to="/home"
                >
                         Go to home 
                </Link>
        </div>
</div>,
      "className": "page-not-found",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="page-not-found__title"
>
            Error 404
</div>,
          <div
            className="page-not-found__content"
>
            Woops. Looks like this page doesn't exist 
</div>,
          <div
            className="page-not-found__link"
>
            <Link
                        replace={false}
                        to="/home"
            >
                         Go to home 
            </Link>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error 404",
            "className": "page-not-found__title",
          },
          "ref": null,
          "rendered": "Error 404",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Woops. Looks like this page doesn't exist ",
            "className": "page-not-found__content",
          },
          "ref": null,
          "rendered": "Woops. Looks like this page doesn't exist ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/home"
>
               Go to home 
</Link>,
            "className": "page-not-found__link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Go to home ",
              "replace": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": " Go to home ",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="page-not-found__title"
          >
                    Error 404
          </div>
          <div
                    className="page-not-found__content"
          >
                    Woops. Looks like this page doesn't exist 
          </div>
          <div
                    className="page-not-found__link"
          >
                    <Link
                              replace={false}
                              to="/home"
                    >
                               Go to home 
                    </Link>
          </div>
</div>,
        "className": "page-not-found",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="page-not-found__title"
>
              Error 404
</div>,
            <div
              className="page-not-found__content"
>
              Woops. Looks like this page doesn't exist 
</div>,
            <div
              className="page-not-found__link"
>
              <Link
                            replace={false}
                            to="/home"
              >
                             Go to home 
              </Link>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Error 404",
              "className": "page-not-found__title",
            },
            "ref": null,
            "rendered": "Error 404",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Woops. Looks like this page doesn't exist ",
              "className": "page-not-found__content",
            },
            "ref": null,
            "rendered": "Woops. Looks like this page doesn't exist ",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/home"
>
                 Go to home 
</Link>,
              "className": "page-not-found__link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Go to home ",
                "replace": false,
                "to": "/home",
              },
              "ref": null,
              "rendered": " Go to home ",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
